mmv1/templates/terraform/pre_create/alloydb_cluster.go.erb

```diff
@@ -30,4 +30,36 @@ if backupSource != nil || continuousBackupSource != nil {
 	}
 	restoreClusterRequestBody["cluster"] = cluster
 	obj = restoreClusterRequestBody
-}
\ No newline at end of file
+}
+
+
+// Read the secondary cluster config to call the api for creating secondary cluster
+
+var secondaryConfig interface{}
+var clusterType interface{}
+
+if val, ok := obj["secondaryConfig"]; ok {
+	secondaryConfig = val
+}
+
+if val, ok := obj["clusterType"]; ok {
+	clusterType = val
+}
+
+if clusterType == "SECONDARY" {
+	if secondaryConfig != nil {
+		// Use createsecondary API if this is a secondary cluster
+		url = strings.Replace(url, "clusters?clusterId", "clusters:createsecondary?cluster_id", 1)
+
+	// Validation error if secondary_config is not defined
+	} else {
+		return fmt.Errorf("Error creating cluster. Can not create secondary cluster without secondary_config field.")
+	}
+}
+
+// Validation error if secondary_config is defined but, cluster type is not secondary
```

Comment:
We could consider moving this validation up to a CustomizeDiff. That will apply it at plan time rather than apply time, making it more useful for end users.

Reply:
Noted as a Todo for later as it should be backward compatible.